packages/back-end/test/data-generator/new-generator.ts

```diff
@@ -37,16 +33,34 @@ type EventTableData = TableData & {
   event: string;
 };
 
-const userRetention: Record<string, number> = {};
+interface DataTables {
+  pageViews: PageViewTableData[];
+  sessions: SessionTableData[];
+  experimentViews: ExperimentTableData[];
+  purchases: PurchaseTableData[];
+  events: EventTableData[];
+  userRetention: Record<string, number>;
+}
+
+interface SimulatorData {
+  dataTables: DataTables;
+  messyData: boolean;
+  expKeyPrefix: string;
+  startDate: Date;
+  currentDate: Date;
+  runLengthDays: int;
+}
 
-// use fixed startDate so that integration tests can filter reliably
-const startDate = new Date();
-startDate.setDate(startDate.getDate() - 90);
-function getDateRangeCondition(start: number, end: number) {
+function getDateRangeCondition(
+  startDate: Date,
+  dateLength: int,
+  startPct: number,
+  endPct: number
+) {
   const s = new Date(startDate);
   const e = new Date(startDate);
-  s.setDate(s.getDate() + start);
-  e.setDate(e.getDate() + end);
+  s.setDate(s.getDate() + startPct * dateLength);
+  e.setDate(e.getDate() + endPct * dateLength);
```

Comment:
We may want to consider using `date-fns` to clean up a lot of the date manipulation in this part of the codebase. Existing example below

https://github.com/growthbook/growthbook/blob/77cc4661830a325bac650a92cffe05f770b46157/packages/back-end/src/integrations/SqlIntegration.ts#L1601

Reply:
I use `addDays` to clean this up in a few places but I didn't want to refactor everything else for this standalone script because much of it relies on in-place manipulations of the date. ty.